lib/pure/parsecfg.nim

```diff
@@ -430,40 +430,52 @@ proc newConfig*(): Config =
   ## Useful when wanting to create a configuration file.
   result = newOrderedTable[string, OrderedTableRef[string, string]]()
 
-proc loadConfig*(filename: string): Config =
-  ## Load the specified configuration file into a new Config instance.
+proc loadConfigFromStream*(stream: Stream, filename: string = "[stream]"): Config =
+  ## Load the specified configuration from stream into a new Config instance.
+  ## `filename` parameter is only used for nicer error messages.
   var dict = newOrderedTable[string, OrderedTableRef[string, string]]()
   var curSection = "" ## Current section,
                       ## the default value of the current section is "",
                       ## which means that the current section is a common
   var p: CfgParser
-  var fileStream = newFileStream(filename, fmRead)
-  if fileStream != nil:
-    open(p, fileStream, filename)
-    while true:
-      var e = next(p)
-      case e.kind
-      of cfgEof:
-        break
-      of cfgSectionStart: # Only look for the first time the Section
-        curSection = e.section
-      of cfgKeyValuePair:
-        var t = newOrderedTable[string, string]()
-        if dict.hasKey(curSection):
-          t = dict[curSection]
-        t[e.key] = e.value
-        dict[curSection] = t
-      of cfgOption:
-        var c = newOrderedTable[string, string]()
-        if dict.hasKey(curSection):
-          c = dict[curSection]
-        c["--" & e.key] = e.value
-        dict[curSection] = c
-      of cfgError:
-        break
-    close(p)
+  open(p, stream, filename)
+  while true:
+    var e = next(p)
+    case e.kind
+    of cfgEof:
+      break
+    of cfgSectionStart: # Only look for the first time the Section
+      curSection = e.section
+    of cfgKeyValuePair:
+      var t = newOrderedTable[string, string]()
+      if dict.hasKey(curSection):
+        t = dict[curSection]
+      t[e.key] = e.value
+      dict[curSection] = t
+    of cfgOption:
+      var c = newOrderedTable[string, string]()
+      if dict.hasKey(curSection):
+        c = dict[curSection]
+      c["--" & e.key] = e.value
+      dict[curSection] = c
+    of cfgError:
+      break
+  close(p)
   result = dict
 
+proc loadConfigFromString*(str: string, filename: string = "[string]"): Config =
```

Comment:
This can be removed because it will be handled by the ``stream`` case (using StringStream).